endpoints/collections/src/main.cpp

```diff
@@ -348,19 +356,62 @@ namespace
 				}
 
 				auto conn = irods::get_connection(client_info.username);
-				fs::client::create_collections(conn, lpath_iter->second);
+				bool created = false;
+				int ec = 0;
+				const auto iter = _args.find("create-intermediates");
+
+				if (iter != std::end(_args) && iter->second == "1") {
+					//
+					// This branch implements a modified version of irods::filesystem::_::create_collections().
+					// These changes may be absorbed into the filesystem library. If that happens, don't forget
+					// to update this code block to use the official implementation.
+					//
+
+					fs::throw_if_path_length_exceeds_limit(lpath_iter->second);
+
+					if (const auto p = fs::path{lpath_iter->second}.parent_path(); !fs::client::exists(conn, p)) {
```

Comment:
This does not work when there are multiple non-existent collections in the path.

Remove or implement this logic in terms of `fs::create_collection` (involves more network activity - not good).

Reply:
Implemented custom version of fsys function, but cannot make the error codes identical for the `create-intermediates=0` and `create-intermediates=1` cases due to how the iRODS server implements the recursive flag.

See test.